python/triton/compiler/code_generator.py

```diff
@@ -927,6 +927,26 @@ def visit_IfExp(self, node):
             else:
                 return self.visit(node.orelse)
 
+    def visit_With(self, node):
+        # Lower `with` statements by constructing context managers and calling their enter/exit hooks
+        # Instantiate each context manager with builder injection
+        cm_list = []
+        for item in node.items:
+            call = item.context_expr
+            fn = self.visit(call.func)
+            args = [self.visit(arg) for arg in call.args]
+            kws = dict(self.visit(kw) for kw in call.keywords)
```

Comment:
Interesting. I take it this doesn't properly handle

```python
with abc:
    return 
```

Reply:
Do you mean something can be returned or just an empty return?